### PR TITLE
[CH] Fix TPCDS q14

### PR DESCRIPTION
**Pull request by liuneng1994**
## What changes were proposed in this pull request?

Fix TPCDS q14

## How was this patch tested?

unit tests

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala [hunk 1]
```diff
@@ -188,10 +187,24 @@ case class TransformPreOverrides(
       )
       (projectExpressions, expressionPos)
     }
+    def getOutput(plan: SparkPlan) : Seq[Attribute] = {
```
Comment (zzcclp) [inline gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala:190]: does this function suit for velox backend? @rui-mo please help to check

File: gluten-core/src/main/scala/io/glutenproject/backendsapi/TransformerApi.scala [hunk 2]
```diff
@@ -63,4 +64,7 @@ trait TransformerApi {
   def getSupportExpressionClassName: util.Set[String] = {
     util.Collections.emptySet()
   }
+  def getAggregateOutput(plan: SparkPlan) : Seq[Attribute] = {
```
Comment (zzcclp) [inline gluten-core/src/main/scala/io/glutenproject/backendsapi/TransformerApi.scala:67]: add a blank line before this

Comment (rui-mo) [inline gluten-core/src/main/scala/io/glutenproject/backendsapi/TransformerApi.scala:67]: Can we use `HashAggregateExec` as input for this function? With `SparkPlan` used, this function looks general and not restricted to the function name `getAggregateOutput`.

File: gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala (hunk 1 above)
Comment (rui-mo) [reply gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala]: I do not see any change of logic to Velox backend, so ti should be OK.

File: gluten-core/src/main/scala/io/glutenproject/backendsapi/TransformerApi.scala (hunk 2 above)
Comment (liuneng1994) [reply gluten-core/src/main/scala/io/glutenproject/backendsapi/TransformerApi.scala]: done

Comment (liuneng1994) [reply gluten-core/src/main/scala/io/glutenproject/backendsapi/TransformerApi.scala]: rename the method name.This function is to get the plan node output

File: gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala [hunk 3]
```diff
@@ -165,7 +164,7 @@ case class TransformPreOverrides(
    * */
   private def addProjectionForShuffleExchange(plan: ShuffleExchangeExec): (Int, Partitioning,
     SparkPlan) = {
-    def selectEpxressions(exprs: Seq[Expression], attributes: Seq[Attribute])
+    def selectExpressions(exprs: Seq[Expression], attributes: Seq[Attribute])
```
Comment (rui-mo) [inline gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala:167]: Nice catch!

**Review by rui-mo — APPROVED** [review]

**Comment by ulysses-you** [comment]
This pr breaks velox CI due to unused import.. I fixed it in https://github.com/oap-project/gluten/pull/2286 together

**Pull request merged**